lib/helpers/createExtensions.js

```diff
@@ -1,24 +1,33 @@
 'use strict';
 
 var uid = 0;
-
+var tasksUid = 0;
 var metadata = require('./metadata');
 
-function Storage(name){
+function Storage(name, tId){
   this.uid = uid++;
+  this.taskId = tId;
   this.name = name;
 }
 
+function taskId(fn) {
+  if (fn._uid == null) {
+    fn._uid = tasksUid++;
```

Comment:
I believe this should be on `meta` instead of the actual function.  I've always been trying to keep the user-defined function untouched.